src/Microsoft.Extensions.Primitives/StringTokenizer.cs

```diff
@@ -21,14 +21,29 @@ public struct StringTokenizer :  IEnumerable<StringSegment>
         /// <param name="value">The <c>string</c> to tokenize.</param>
         /// <param name="separator">The character to tokenize by.</param>
         public StringTokenizer(string value, char separator)
```

Comment:
Could we remove this? It's an extra array allocation that could be cached by the caller instead.